packages/framework/src/shared/common.ts

```diff
@@ -163,11 +163,12 @@ export const error = (value: unknown): void => {
 export const replace =
   (replacer: (expr: string) => string) =>
   (val: string): unknown => {
-    const matches = val.match(/\$\{+[^}]+\}+/g);
+    const replaceRegex = /\$\{(?:[^}{]+|\{(?:[^}{]+|\{[^}{]*\})*\})*\}/g;
```

Comment:
How did you come up with this regex? This is going to take anyone a long time to understand. Please leave a comment and tests to describe what this does.

Consider replacing this regex with a stack implementation instead since regex is infamously not meant for capturing things like matching parens/braces.